src/uncurryN.js

```diff
@@ -0,0 +1,43 @@
+var _curry2 = require('./internal/_curry2');
+var _slice = require('./internal/_slice');
+var arity = require('./arity');
+var is = require('./is');
+
+
+/**
+ * Returns a function of arity `n` from a (manually) curried function.
+ *
+ * @func
+ * @memberOf R
+ * @category Function
+ * @sig Number -> (a -> b) -> (a -> c)
+ * @param {Number} length The arity for the returned function.
+ * @param {Function} fn The function to uncurry.
+ * @return {Function} A new function.
+ * @see R.curry
+ * @example
+ *
+ *      var addFour = function(a) {
+ *        return function(b) {
+ *          return function(c) {
+ *            return function(d) {
+ *              return a + b + c + d;
+ *            };
+ *          };
+ *        };
+ *      };
+ *
+ *      var uncurriedAddFour = R.uncurryN(4, addFour);
+ *      curriedAddFour(1, 2, 3, 4); //=> 10
+ */
+module.exports = _curry2(function uncurryN(length, fn) {
+  return arity(length, function() {
+    var idx = 0;
+    var value = fn;
+    while (idx < length && is(Function, value)) {
```

Comment:
`recurry` is fine, though I think `uncurry` is fine too. `curry` returns a function which accepts all its arguments at once, so I have no problem with `uncurry` returning a function which accepts its arguments one by one.